cassiopeia/tools/branch_length_estimator/BranchLengthEstimator.py

```diff
@@ -0,0 +1,26 @@
+"""
+Abstract class BranchLengthEstimator, for the branch length estimation module.
+
+All algorithms are derived classes of this abstract class, and at a minimum
+implement a method called `estimate_branch_lengths`.
+"""
+import abc
+
+from cassiopeia.data import CassiopeiaTree
+
+
+class BranchLengthEstimator(abc.ABC):
+    """
+    BranchLengthEstimator is an abstract class that all branch length
+    estimation algorithms derive from. At minimum, all BranchLengthEstimator
+    subclasses will implement a method called `estimate_branch_lengths`.
+    """
+
```

Comment:
Are there going to be any general variables for the BranchLengthEstimator class such that we want to add this to an abstract `__init__`?

Reply:
I don't think so, at least not right now. I'd say if any repeated code comes up in the future that is clearly an essential part of a BranchLengthEstimator's init we can add it at that time.

Reply:
Great, let's proceed as you suggest.